### ISSUE TITLE
Client rare crash: 

### ISSUE THREAD

**Issue by bradyrussell** [bradyrussell/marchingcubesterrainplugin#41]
```
[2020.05.05-12.59.05:671][815]LogWindows: Could not start crash report client using ../../../Engine/Binaries/Win64/CrashReportClientEditor-Win64-Development.exe
[2020.05.05-12.59.05:671][815]LogMemory: Platform Memory Stats for Windows
[2020.05.05-12.59.05:671][815]LogMemory: Process Physical Memory: 2770.79 MB used, 2991.35 MB peak
[2020.05.05-12.59.05:671][815]LogMemory: Process Virtual Memory: 3328.02 MB used, 3568.44 MB peak
[2020.05.05-12.59.05:671][815]LogMemory: Physical Memory: 14550.43 MB used,  18130.03 MB free, 32680.46 MB total
[2020.05.05-12.59.05:671][815]LogMemory: Virtual Memory: 9156.07 MB used,  134208568.00 MB free, 134217728.00 MB total
[2020.05.05-12.59.05:671][815]LogThreadingWindows: Error: Runnable thread PoolThread 30 crashed.
[2020.05.05-12.59.05:671][815]LogWindows: Error: === Critical error: ===
[2020.05.05-12.59.05:671][815]LogWindows: Error: 
[2020.05.05-12.59.05:671][815]LogWindows: Error: Fatal error!
[2020.05.05-12.59.05:671][815]LogWindows: Error: 
[2020.05.05-12.59.05:671][815]LogWindows: Error: Unhandled Exception: 0xe06d7363
[2020.05.05-12.59.05:671][815]LogWindows: Error: 
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff8913b9709 KERNELBASE.dll!UnknownFunction []
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff87fde4890 VCRUNTIME140.dll!UnknownFunction []
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff8100eec1d UE4Editor-MCTPlugin.dll!UnknownFunction []
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff81011a928 UE4Editor-MCTPlugin.dll!UnknownFunction []
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff81011b08c UE4Editor-MCTPlugin.dll!UnknownFunction []
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff8100eb0d2 UE4Editor-MCTPlugin.dll!UnknownFunction []
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff8100ecfc2 UE4Editor-MCTPlugin.dll!UnknownFunction []
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff8100ecce3 UE4Editor-MCTPlugin.dll!UnknownFunction []
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff8100fba2b UE4Editor-MCTPlugin.dll!UnknownFunction []
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff8100f9d97 UE4Editor-MCTPlugin.dll!UnknownFunction []
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff829b6eb85 UE4Editor-Core.dll!FQueuedThread::Run() [f:\ue4_24\ue4_24\engine\source\runtime\core\private\hal\threadingbase.cpp:530]
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff829ece37f UE4Editor-Core.dll!FRunnableThreadWin::Run() [f:\ue4_24\ue4_24\engine\source\runtime\core\private\windows\windowsrunnablethread.cpp:96]
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff829ec5591 UE4Editor-Core.dll!FRunnableThreadWin::GuardedRun() [f:\ue4_24\ue4_24\engine\source\runtime\core\private\windows\windowsrunnablethread.cpp:53]
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff892ad7974 KERNEL32.DLL!UnknownFunction []
[2020.05.05-12.59.05:671][815]LogWindows: Error: [Callstack] 0x00007ff89483a271 ntdll.dll!UnknownFunction []
[2020.05.05-12.59.05:671][815]LogWindows: Error: 
[2020.05.05-12.59.05:671][815]LogWindows: Error: Crash in runnable thread PoolThread 30
```

What thread pool task on the client would crash? Extraction maybe?

**Comment by bradyrussell** [comment]
Maybe polyvox threw an exception?

**Comment by bradyrussell** [comment]
This may be the exception from #29

**Comment by bradyrussell** [comment]
[2020.05.11-23.46.44:087][363]LogAudio: Display: Waited 46.853699 ms for audio thread.
[2020.05.11-23.46.52:948][957]LogVoxelWorld: Warning: Client tried to access nonexistant region X=128 Y=128 Z=-544.
[2020.05.11-23.46.52:948][957]LogVoxelWorld: Warning: Non-air region was meshed but actor has not replicated.
[2020.05.11-23.46.52:948][957]LogVoxelWorld: Warning: Client tried to access nonexistant region X=96 Y=160 Z=-544.
[2020.05.11-23.46.52:948][957]LogVoxelWorld: Warning: Non-air region was meshed but actor has not replicated.
[2020.05.11-23.46.52:948][957]LogVoxelWorld: Warning: Client tried to access nonexistant region X=192 Y=128 Z=-544.
[2020.05.11-23.46.52:948][957]LogVoxelWorld: Warning: Non-air region was meshed but actor has not replicated.
[2020.05.11-23.47.08:146][535]LogVoxelWorld: Error: [Error] MarchingCubesExtractionTask caught exception extracting [X=160 Y=-32 Z=-736] [No space in chunk array for new chunk.].
[2020.05.11-23.47.08:148][536]LogVoxelWorld: Error: [Error] MarchingCubesExtractionTask caught exception extracting [X=128 Y=128 Z=-736] [No space in chunk array for new chunk.].

**Comment by bradyrussell** [comment]
Not so rare anymore. Traveling any moderate distance like 500 meters seems to make it likely to happen.

**Comment by bradyrussell** [comment]
Not fixed but closing because it is definitely #52
